pkg/codegen/schema/schema.go

```diff
@@ -1186,26 +1186,59 @@ func (t *types) bindEnumTypeDetails(enum *EnumType, token string, spec ComplexTy
 	if err != nil {
 		return err
 	}
+
+	values, err := t.bindEnumValues(spec.Enum, typ)
+	if err != nil {
+		return err
+	}
+
 	enum.Token = token
-	enum.Elements = t.bindEnumValues(spec.Enum)
+	enum.Elements = values
 	enum.ElementType = typ
 	enum.Comment = spec.Description
 
 	return nil
 }
 
-func (t *types) bindEnumValues(values []*EnumValueSpec) (enums []*Enum) {
+func (t *types) bindEnumValues(values []*EnumValueSpec, typ Type) ([]*Enum, error) {
+	var enums []*Enum
+
+	errorMessage := func(val interface{}, expectedType string) error {
+		return fmt.Errorf("cannot assign enum value of type '%T' to enum of type '%s'", val, expectedType)
+	}
 	for _, spec := range values {
+		switch typ {
+		case StringType:
+			if _, ok := spec.Value.(string); !ok {
+				return enums, errorMessage(spec.Value, typ.String())
+			}
+		case IntType:
+			v, ok := spec.Value.(float64)
```

Comment:
Is `float64` correct here?

Reply:
Yeah, Unmarshal decodes [json numbers to float64](https://golang.org/pkg/encoding/json/#Unmarshal:~:text=bool%2C%20for%20JSON%20booleans,nil%20for%20JSON%20null) when unmarshalling to an interface. So we have to do the additional check to see if it's a true int.

That being said, your comment did point out that I was missing some pieces which I fixed up and added some more test details.